### PR TITLE
Foundation overlay needs to build with -swift-version 5

### PR DESCRIPTION
**Pull request by Catfish-Man**
The Foundation overlay still builds in Swift 3 mode, which is going away. We should fix that!

### CODE REVIEW

**Comment by Catfish-Man** [comment]
@swift-ci please test

**Review by itaiferber — APPROVED** [review]
Looks good!

File: stdlib/public/SDK/Foundation/Data.swift [hunk 1]
```diff
@@ -664,6 +678,8 @@ internal final class _DataStorage {
             _backing = .customMutableReference(data)
         case .customMutableReference(let d):
             d.replaceBytes(in: range, withBytes: replacementBytes, length: replacementLength)
+        @unknown default:
+            fatalError("Unknown Data backing type")
```
Comment (jrose-apple) [inline stdlib/public/SDK/Foundation/Data.swift]: You're going to have to come back and do something about these. If Data really can grow new representations, you'll need to make sure they're handled even in inlined code. Sadly we already have a pattern for this: https://twitter.com/UINT_MIN/status/1020503397357760512

Comment (Catfish-Man) [reply stdlib/public/SDK/Foundation/Data.swift]: Argh :( ok, I should talk to @phausler about this

Comment (phausler) [reply stdlib/public/SDK/Foundation/Data.swift]: this code is going to have some changes coming in soon anyhow.

**Review by phausler — APPROVED** [review]
the rest of this looks good to me.

**Pull request merged**